tests/integration/coordinator/plugins/clickhouse/test_plugin.py

```diff
@@ -343,6 +430,57 @@ async def test_restores_materialized_view_data(restored_cluster: Sequence[ClickH
         assert response == expected_data
 
 
+async def test_restores_materialized_view_deleted_source_table(restored_cluster: Sequence[ClickHouseClient]) -> None:
+    s1_data = [[7 * 3]]
+    s2_data = [[9 * 3]]
+    cluster_data = [s1_data, s1_data, s2_data]
+    for client, expected_data in zip(restored_cluster, cluster_data):
+        response = await client.execute(b"SELECT thekey3 FROM default.materialized_view_deleted_source ORDER BY thekey3")
+        assert response == expected_data
+
+
+async def test_restores_materialized_view_with_undeleted_source_table(
+    function_restored_cluster: Sequence[ClickHouseClient],
+) -> None:
+    await function_restored_cluster[0].execute(
+        b"CREATE TABLE default.source_table_for_view_deleted (thekey UInt32, thedata String) "
+        b"ENGINE = ReplicatedMergeTree ORDER BY (thekey)"
+    )
+    await function_restored_cluster[0].execute(b"INSERT INTO default.source_table_for_view_deleted VALUES (8, '8')")
+    s1_data = [[7 * 3], [8 * 3]]
+    s2_data = [[9 * 3]]
+    cluster_data = [s1_data, s1_data, s2_data]
+    # Recreated deleted table works again
+    for client, expected_data in zip(function_restored_cluster, cluster_data):
+        response = await client.execute(b"SELECT thekey3 FROM default.materialized_view_deleted_source ORDER BY thekey3")
+        assert response == expected_data
+
+
+async def test_restores_view_with_deleted_source_table(restored_cluster: Sequence[ClickHouseClient]) -> None:
+    unknown_table_exception_code = 60
+    for client in restored_cluster:
+        with pytest.raises(ClickHouseClientQueryError) as raised:
+            await client.execute(b"SELECT thekey2 FROM default.view_deleted_source ORDER BY thekey2")
+        assert raised.value.status_code == 404
+        assert raised.value.exception_code == unknown_table_exception_code
+
+
+async def test_restores_view_undeleted_source_table(restored_cluster: Sequence[ClickHouseClient]) -> None:
```

Comment:
The other test is using `function_restored_cluster`, but this test, which contains CREATE TABLE, still uses the global `restored_cluster`.

Since it's mutating, it should also use `function_restored_cluster` ?